Source/ACE.Server/Factories/LootGenerationFactory.cs

```diff
@@ -35,5 +44,18050 @@ public static List<WorldObject> CreateRandomObjectsOfType(WeenieType type, int c
 
             return worldObjects;
         }
+
+        public static void CreateRandomTestWorldObjects(Player player, uint typeId, uint numItems)
+        {
+            throw new System.NotImplementedException();/*
+            var weenieList = DatabaseManager.World.GetRandomWeeniesOfType(typeId, numItems);
+            List<WorldObject> items = new List<WorldObject>();
+            for (int i = 0; i < numItems; i++)
+            {
+                WorldObject wo = WorldObjectFactory.CreateNewWorldObject(weenieList[i].WeenieClassId);
+                items.Add(wo);
+            }
+            player.HandleAddNewWorldObjectsToInventory(items);*/
+        }
+
+        public static WorldObject CreateRandomLootObjects(int tier)
+        {
+            WorldObject wo;
+            switch (tier)
+            {
+                case 1:
+                    //loot tier 1
+                    wo = CreateTier1Objects();
+                    break;
+                case 2:
+                    wo = CreateTier2Objects();
+                    //loot tier 2
+                    break;
+                case 3:
+                    //loot tier 3
+                    wo = CreateTier3Objects();
+                    break;
+                case 4:
+                    //loot tier 4
+                    wo = CreateTier4Objects();
+                    break;
+                case 5:
+                    //loot tier 5
+                    wo = CreateTier5Objects();
+                    break;
+                case 6:
+                    //loot tier 6
+                    wo = CreateTier6Objects();
+                    break;
+                case 7:
+                    //loot tier 7
+                    wo = CreateTier7Objects();
+                    break;
+                default:
+                    //loot tier 8
+                    wo = CreateTier8Objects();
+                    break;
+            }
+            return wo;
+        }
+
+        public static WorldObject CreateTier1Objects()
+        {
+            WorldObject wo;
+            int id = 0;
+            int chance;
+            Random r = new Random();
```

Comment:
Depending on how quick this new Random() line is called, creating new instances of Random could raise some issues

Once in a previous project I was creating a new instance of Random everytime i generated a new random #, and it turned out to be a big issue. If the code was called fast enough, basically it would return the same random # a bunch of times in a row

Might not be an issue here, but it's something to be aware of for sure!

In the Physics.Common namespace there is a Random class that has a static instance of System.Random, and it is marked with the [ThreadStatic] attribute in an attempt to make it thread safe.

Reply:
Locking variables is also costly, and blocks other threads as well.

In this case I'd just instantiate a new random every time like he has it here. If, after profiling, we determine it is an issue, we could then optimize it by having a Random object per landblock and passing that as an argument, but, I don't think that will be necessary.

Reply:
I think the behavior @gmriggs saw is actually in full effect in our encounter system currently. I've found often enough that it irks me that if I load into a landblock and a rare mob is selected, you'll find that same mob spawned in several generators. Certainly, it feels like something is off there.